Translation of Conditional Rendering (#22)
reactjs/ca.reactjs.org: content/docs/conditional-rendering.md

---
id: conditional-rendering
title: Renderitzat condicional
permalink: docs/conditional-rendering.html
prev: handling-events.html
next: lists-and-keys.html
redirect_from:
  - "tips/false-in-jsx.html"
---

A React, pots crear diferents components que encapsulin el comportament que necessitis. Llavors, pots renderitzar només alguns d'ells, depenent de l'estat de la teva aplicació.

> El renderitzat condicional a React funciona de la mateixa forma que les condicions funcionen a JavaScript. Usa operadors de JavaScript com [`if`](https://developer.mozilla.org/ca/docs/Web/JavaScript/Reference/Statements/if...else) o l'[operador condicional](https://developer.mozilla.org/ca/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) per crear elements que representin l'estat actual, i deixa que React actualitzi la interfície d'usuari per emparellar-los.

Reviewer: ```suggestion
El renderitzat condicional a React funciona de la mateixa manera que les condicions funcionen a JavaScript. Usa operadors de JavaScript com [`if`](https://developer.mozilla.org/ca/docs/Web/JavaScript/Reference/Statements/if...else) o l'[operador condicional](https://developer.mozilla.org/ca/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) per crear elements que representin l'estat actual, i deixa que React actualitzi la interfície d'usuari per emparellar-los.
```
Forma tambè és correcte

}
```

> Crearem un component `Greeting` que mostra qualsevol d'aquests dos components depenent de si un usuari ha iniciat sessió:

Reviewer: ```suggestion
Crearem un component `Greeting` que mostri qualsevol d'aquests dos components depenent de si un usuari ha iniciat sessió:
```

);
```

[**Prova-ho a CodePen**](https://codepen.io/gaearon/pen/QKzAgB?editors=0010)

> Tot i que declarar una variable i usar una sentència `if` és una bona forma de renderitzar condicionalment un component, a vegades pot ser que vulguis usar una altra sintaxi. Hi ha unes quantes formes de fer condicions en línia amb JSX, explicades a sota.

Reviewer: ```suggestion
Tot i que declarar una variable i usar una sentència `if` és una bona manera de renderitzar condicionalment un component, a vegades pot ser que vulguis fer servir una altra sintaxi. Hi ha unes quantes formes de fer condicions en línia amb JSX, explicades a continuació.
```

}
```

> De la mateixa forma que a JavaScript, tu ets qui decideixes usar un estil apropiat basat en el qual tu i el teu equip considereu més llegible. També recorda que sempre que les condicions es tornen massa complexes, pot ser un bon moment per [extreure un component](/docs/components-and-props.html#extracting-components).

Reviewer: ```suggestion
De la mateixa forma que a JavaScript, tu ets qui decideixes usar un estil apropiat basat en el qual tu i el teu equip considereu més llegible. Recorda que quan les condicions es tornin massa complexes, pot ser un bon moment per [extreure un component](/docs/components-and-props.html#extracting-components).
```
